src/rust/graph-query/src/var_allocator.rs

```diff
@@ -35,12 +38,9 @@ impl VarAllocator {
         variables
     }
 
-    pub fn variable_map(self) -> std::collections::HashMap<String, String> {
-        let mut map = std::collections::HashMap::with_capacity(self.variables.len());
-        for (k, v) in self.variables.into_iter() {
-            map.insert(v, k);
-        }
-        map
+    pub fn variable_map(self) -> HashMap<String, String> {
+        // The dgraph-tonic client expects a std HashMap, so we can't use our fnv map
+        HashMap::from_iter(self.variables.into_iter().map(|(k, v)| (v, k)))
```

Comment:
I totally missed the swapping of key and value initially; why is that done? Seems worth some documentation.

Reply:
I've added an additional note to the comment.